src/extractor/guidance/intersection_handler.cpp

```diff
@@ -383,136 +393,140 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
-        {
-            for (std::size_t i = 1; i < intersection.size(); ++i)
-            {
-                const auto &road = intersection[i];
-                const auto &road_data = node_based_graph.GetEdgeData(road.eid);
-
-                const auto same_name = !util::guidance::requiresNameAnnounced(
-                    in_data.name_id, road_data.name_id, name_table, street_name_suffix_table);
+    // best_continue is the first sorted way that shares the same name as the in way
+    const auto sameName = [&](const out_way &lhs) { return lhs.same_name_id == true; };
+    const auto best_continue_it = std::find_if(begin(out_ways), End, sameName);
+    if (best_continue_it != End)
+    {
+        best_continue = best_continue_it->index;
+        best_continue_deviation = best_continue_it->deviation_from_straight;
+    }
 
-                if (same_name)
-                {
-                    ++count;
-                    if (road.entry_allowed)
-                        ++count_valid;
-                }
-            }
-        }
-        return std::make_pair(count, count_valid);
-    }();
+    const auto &best_data = node_based_graph.GetEdgeData(intersection[best].eid);
 
+    // if the best angle is going straight but the road is turning, declare no obvious turn
     if (0 != best_continue && best != best_continue &&
-        angularDeviation(intersection[best].angle, STRAIGHT_ANGLE) <
-            MAXIMAL_ALLOWED_NO_TURN_DEVIATION &&
+        best_deviation < MAXIMAL_ALLOWED_NO_TURN_DEVIATION &&
         node_based_graph.GetEdgeData(intersection[best_continue].eid).road_classification ==
-            node_based_graph.GetEdgeData(intersection[best].eid).road_classification)
+            best_data.road_classification)
     {
-        // if the best angle is going straight but the road is turning, we don't name anything
-        // obvious
         return 0;
     }
 
+    // checks if continue candidates are sharp turns
     const bool all_continues_are_narrow = [&]() {
-        if (in_data.name_id == EMPTY_NAMEID)
-            return false;
-
-        return std::count_if(
-                   intersection.begin() + 1, intersection.end(), [&](const ConnectedRoad &road) {
-                       const auto &road_data = node_based_graph.GetEdgeData(road.eid);
-                       const auto same_name =
-                           !util::guidance::requiresNameAnnounced(in_data.name_id,
-                                                                  road_data.name_id,
-                                                                  name_table,
-                                                                  street_name_suffix_table);
-
-                       return same_name &&
-                              angularDeviation(road.angle, STRAIGHT_ANGLE) < NARROW_TURN_ANGLE;
-                   }) == num_continue_names.first;
+        return std::count_if(begin(out_ways), end(out_ways), [&](const out_way &road) {
```

Comment:
This does not need to be wrapped in a lambda. You can write `const bool all_continues_are_narrow = std::count_if(...) == num_continue_names.first`.